From 44eedec0d79b9f36733962a5cfdd677ca7fe1d5f Mon Sep 17 00:00:00 2001
From: Steven Walter <swalter@lexmark.com>
Date: Mon, 27 Apr 2015 20:55:13 -0400
Subject: [PATCH] Unconditionally depend on openssl-sys

---
 libssh2-sys/Cargo.toml | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/libssh2-sys/Cargo.toml b/libssh2-sys/Cargo.toml
index 8e27a71..1edbf65 100644
--- a/libssh2-sys/Cargo.toml
+++ b/libssh2-sys/Cargo.toml
@@ -15,24 +15,6 @@ path = "lib.rs"
 [dependencies]
 libz-sys = "0.1.0"
 libc = "0.1"
-
-[target.i686-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
-[target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
-[target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-[target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-[target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-[target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.6.0"
-[target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
-[target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
-[target.x86_64-unknown-dragonfly.dependencies]
 openssl-sys = "0.6.0"
 
 [build-dependencies]
-- 
1.9.1

