From eee79cc95f57434c4b530da96c88494e9b5d056b Mon Sep 17 00:00:00 2001
From: Steven Walter <stevenrwalter@gmail.com>
Date: Sat, 27 Aug 2016 13:20:48 -0400
Subject: [PATCH] Don't build unittests

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 2b9468a..2bb30f6 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ ifeq ($(BUILD_DIRS_ONLY),1)
   OPTIONAL_DIRS := tools/clang/utils/TableGen
 else
   DIRS := lib/Support lib/TableGen utils lib/IR lib tools/llvm-shlib \
-          tools/llvm-config tools docs cmake unittests
+          tools/llvm-config tools docs cmake
   OPTIONAL_DIRS := projects bindings
 endif
 
@@ -39,7 +39,7 @@ ifeq ($(BUILD_EXAMPLES),1)
   OPTIONAL_DIRS += examples
 endif
 
-EXTRA_DIST := test unittests llvm.spec include win32 Xcode
+EXTRA_DIST := test llvm.spec include win32 Xcode
 
 include $(LEVEL)/Makefile.config
 
-- 
2.7.4

