From 9652ddba460f30e83f401ab1564656e7787bdea9 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Wed, 3 Feb 2016 15:59:48 -0500
Subject: [PATCH] disable cargo snapshot fetch

---
 Makefile.in | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 286a593..9c66486 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -92,10 +92,6 @@ test-unit-$(1): $$(CARGO)
 endef
 $(foreach target,$(CFG_TARGET),$(eval $(call CARGO_TARGET,$(target))))
 
-$(TARGET_ROOT)/snapshot/bin/cargo$(X): src/snapshots.txt
-	$(CFG_PYTHON) src/etc/dl-snapshot.py $(CFG_BUILD)
-	touch $@
-
 
 # === Tests
 
-- 
2.7.0

