From 66911257957f29ccf3a4386956977e97168200e7 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Tue, 25 Nov 2014 16:28:25 -0500
Subject: [PATCH] XXX: hacks

---
 src/lib.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib.rs b/src/lib.rs
index de90de6..a388af3 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -99,13 +99,14 @@ pub fn compile_library(output: &str, config: &Config, files: &[&str]) {
 }
 
 fn run(cmd: &mut Command) {
-    println!("running: {}", cmd);
+    println!("running: '{}' : {}", cmd.program(), cmd);
     assert!(cmd.stdout(InheritFd(1))
                .stderr(InheritFd(2))
                .status()
                .unwrap()
                .success());
 
+    println!("Run done");
 }
 
 fn gcc() -> String {
-- 
2.1.3

