From 33ed066e368b4d87bfabbfdb64f4fbcd8821a2ff Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Sat, 15 Nov 2014 14:16:39 -0500
Subject: [PATCH] get-snapshot-debug

---
 src/etc/get-snapshot.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/etc/get-snapshot.py b/src/etc/get-snapshot.py
index 886a84b..6043ee2 100755
--- a/src/etc/get-snapshot.py
+++ b/src/etc/get-snapshot.py
@@ -35,6 +35,7 @@ def unpack_snapshot(triple, dl_path):
     print("extracting " + p)
     tar.extract(p, download_unpack_base)
     tp = os.path.join(download_unpack_base, p)
+    print("Path is " + tp + "; fp = " + fp)
     if os.path.isdir(tp) and os.path.exists(fp):
         continue
     shutil.move(tp, fp)
-- 
2.0.4

