From 7200c63876da23634fea0c0987e4f955ebd08baf Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Thu, 13 Nov 2014 17:14:46 -0500
Subject: [PATCH] src/etc/snapshot: support triples lacking a vendor

---
 src/etc/snapshot.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/etc/snapshot.py b/src/etc/snapshot.py
index e0610bf..85798da 100644
--- a/src/etc/snapshot.py
+++ b/src/etc/snapshot.py
@@ -75,7 +75,11 @@ def full_snapshot_name(date, rev, platform, hsh):
 
 
 def get_kernel(triple):
-    os_name = triple.split('-')[2]
+    t=triple.split('-')
+    if len(t) == 2:
+      os_name = triple.split('-')[1]
+    else:
+      os_name = triple.split('-')[2]
     if os_name == "windows":
         return "winnt"
     if os_name == "darwin":
-- 
2.1.3

