From 5ac7fc317ed975ebdf86f222fdecec58ed2addd2 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 1 Dec 2014 15:50:13 -0500
Subject: [PATCH 02/13] std/io/process: add Debug tests

---
 src/libstd/old_io/process.rs | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/libstd/old_io/process.rs b/src/libstd/old_io/process.rs
index f543e0c..7024b95 100644
--- a/src/libstd/old_io/process.rs
+++ b/src/libstd/old_io/process.rs
@@ -1261,4 +1261,16 @@ mod tests {
         let val = env.get(&EnvKey(CString::new("PATH").unwrap()));
         assert!(val.unwrap() == &CString::new("bar").unwrap());
     }
+
+    fn check_debug<D : fmt::Debug>(c: D, v: &str) {
+        assert_eq!(format!("{:?}", c), v)
+    }
+
+    #[test]
+    fn debug() {
+        check_debug(Command::new("gcc ").arg("-Ifoo'bar"), "'gcc ' '-Ifoo'\\''bar'");
+        check_debug(Command::new("c99"), "'c99'");
+        check_debug(Command::new("c99 "), "'c99 '");
+        check_debug(Command::new("Can't buy me love"), "'Can'\\''t buy me love'");
+    }
 }
-- 
1.9.1

