From c227b05fab797ec7c03ffafea171af0d46a928ff Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 1 Dec 2014 15:50:13 -0500
Subject: [PATCH 02/10] std/io/process: add Debug tests

---
 src/libstd/old_io/process.rs | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/libstd/old_io/process.rs b/src/libstd/old_io/process.rs
index fe12546..ae61c01 100644
--- a/src/libstd/old_io/process.rs
+++ b/src/libstd/old_io/process.rs
@@ -1254,4 +1254,16 @@ mod tests {
         let val = env.get(&EnvKey(CString::new(b"PATH").unwrap()));
         assert!(val.unwrap() == &CString::new(b"bar").unwrap());
     }
+
+    fn check_debug<D : fmt::Debug>(c: D, v: &str) {
+        assert_eq!(format!("{:?}", c), v)
+    }
+
+    #[test]
+    fn debug() {
+        check_debug(Command::new("gcc ").arg("-Ifoo'bar"), "'gcc ' '-Ifoo'\\''bar'");
+        check_debug(Command::new("c99"), "'c99'");
+        check_debug(Command::new("c99 "), "'c99 '");
+        check_debug(Command::new("Can't buy me love"), "'Can'\\''t buy me love'");
+    }
 }
-- 
2.3.1

