From 00dc8106736c6533410fdec0ff0d06fae6ecb18b Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 17 Nov 2014 16:14:15 -0500
Subject: [PATCH] mk/rt/compiler_rt: pass LDFLAGS from CFG_GCCISH_LINK_FLAGS

---
 mk/rt.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mk/rt.mk b/mk/rt.mk
index 7204808..0ef0786 100644
--- a/mk/rt.mk
+++ b/mk/rt.mk
@@ -231,6 +231,7 @@ $$(COMPRT_LIB_$(1)): $$(COMPRT_DEPS) $$(MKFILE_DEPS)
 		AR="$$(AR_$(1))" \
 		RANLIB="$$(AR_$(1)) s" \
 		CFLAGS="$$(CFG_GCCISH_CFLAGS_$(1))" \
+		LDFLAGS="$$(CFG_GCCISH_LINK_FLAGS_$(1))" \
 		TargetTriple=$(1) \
 		triple-builtins
 	$$(Q)cp $$(COMPRT_BUILD_DIR_$(1))/triple/builtins/libcompiler_rt.a $$(COMPRT_LIB_$(1))
-- 
2.1.3

