From 2c23c8dba0769ad0e9c4b0a1130410fdccbf8671 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 9 Mar 2015 17:29:02 -0400
Subject: [PATCH 14/14] filesearch: fix for new path

---
 src/librustc/metadata/filesearch.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/librustc/metadata/filesearch.rs b/src/librustc/metadata/filesearch.rs
index d6dc516..3946376 100644
--- a/src/librustc/metadata/filesearch.rs
+++ b/src/librustc/metadata/filesearch.rs
@@ -217,6 +217,11 @@ pub fn get_or_default_sysroot() -> PathBuf {
             while rel.pop() {
                 p.pop();
             }
+
+            // The last element in a relative path can't be popped, so remove it here.
+            if let Some(_) = p.file_name() {
+                p.pop();
+            }
             p
         }
         None => panic!("can't determine value for sysroot")
-- 
2.3.1

