From c18e65a1b96d12e88bb9068f5ca25c335a910ab2 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Tue, 18 Nov 2014 14:02:36 -0500
Subject: [PATCH] configure: CFG_*DIR_RELATIVE: avoid requiring existence

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index c9bec44..7914601 100755
--- a/configure
+++ b/configure
@@ -566,8 +566,8 @@ fi
 
 # Determine libdir and bindir relative to prefix
 step_msg "calculating relative paths to prefix = ${CFG_PREFIX}"
-CFG_BINDIR_RELATIVE=$(realpath --relative-to="${CFG_PREFIX}" "${CFG_BINDIR}")
-CFG_LIBDIR_RELATIVE=$(realpath --relative-to="${CFG_PREFIX}" "${CFG_LIBDIR}")
+CFG_BINDIR_RELATIVE=$(realpath -m --relative-to="${CFG_PREFIX}" "${CFG_BINDIR}")
+CFG_LIBDIR_RELATIVE=$(realpath -m --relative-to="${CFG_PREFIX}" "${CFG_LIBDIR}")
 
 # Validate Options
 step_msg "validating $CFG_SELF args"
-- 
2.1.3

