From d71d6fa0b9be06cd84eb6c8f7411c624725431c4 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Tue, 25 Nov 2014 12:26:48 -0500
Subject: [PATCH 2/2] openssl-sys is used in curl-rust, so include it
 explicitly here

---
 Cargo.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index ba6e748..9e25cd1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,3 +10,6 @@ git = "https://github.com/servo/rust-url"
 [dependencies.curl-sys]
 path = "curl-sys"
 version = "*"
+
+[dependencies.openssl-sys]
+git = "https://github.com/sfackler/rust-openssl"
-- 
2.1.3

