From 910c8fe2c439ed79451e5bcfb01dedbd4c8225cb Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Tue, 25 Nov 2014 12:26:48 -0500
Subject: [PATCH 2/2] remove per-triple deps on openssl-sys

---
 Cargo.toml | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 1554aea..3a6efa3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,6 +10,7 @@ description = "Rust bindings to libcurl for making HTTP requests"
 [dependencies]
 url = "0.2.0"
 libc = "0.1"
+openssl-sys = "0.5.1"
 
 [dependencies.curl-sys]
 path = "curl-sys"
@@ -17,21 +18,3 @@ version = "0.1.0"
 
 [dev-dependencies]
 log = "0.2"
-
-# Unix platforms use OpenSSL for now to provide SSL functionality
-[target.i686-apple-darwin.dependencies]
-openssl-sys = "0.5.1"
-[target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.5.1"
-[target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.5.1"
-[target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.5.1"
-[target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.5.1"
-[target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.5.1"
-[target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.5.1"
-[target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.5.1"
-- 
2.3.3

